data/sql_setup/prepare_schedule_b.sql

```diff
@@ -18,49 +18,86 @@ create index on ofec_sched_b_queue_old (timestamp);
 create index on ofec_sched_b_queue_new (two_year_transaction_period);
 create index on ofec_sched_b_queue_old (two_year_transaction_period);
 
--- Create trigger to maintain Schedule B queues
-create or replace function ofec_sched_b_update_queues() returns trigger as $$
+-- Create trigger to maintain Schedule A queues for inserts and updates
+-- These happen after a row is inserted/updated so that we can leverage pulling
+-- the new record information from the view itself, which contains the data in
+-- the structure that our tables expect it to be in.
+create or replace function ofec_sched_b_insert_update_queues() returns trigger as $$
 declare
     start_year int = TG_ARGV[0]::int;
     timestamp timestamp = current_timestamp;
     two_year_transaction_period_new smallint;
     two_year_transaction_period_old smallint;
+    view_row fec_vsum_sched_b_vw%ROWTYPE;
 begin
     if tg_op = 'INSERT' then
-        two_year_transaction_period_new = get_transaction_year(new.disb_dt, new.rpt_yr);
+        select into view_row * from fec_vsum_sched_b_vw where sub_id = new.sub_id;
+        two_year_transaction_period_new = get_transaction_year(new.disb_dt, view_row.rpt_yr);
 
         if two_year_transaction_period_new >= start_year then
-            delete from ofec_sched_b_queue_new where sub_id = new.sub_id;
-            insert into ofec_sched_b_queue_new values (new.*, timestamp, two_year_transaction_period_new);
+            delete from ofec_sched_b_queue_new where sub_id = view_row.sub_id;
```

Comment:
All good here? :-)  The setup hasn't changed from before, just how the data is supplied to the queue.